datadog_lambda/tracing.py

```diff
@@ -1320,6 +1329,10 @@ def create_inferred_span_from_eventbridge_event(event, context):
     if span:
         span.set_tags(tags)
     span.start = dt.replace(tzinfo=timezone.utc).timestamp()
+
+    # Since inferred span will later parent Lambda, preserve Lambda's current parent
+    span.parent_id = dd_trace_context.span_id
```

Comment:
It seems that `dd_trace_context` would always have span_id. But, in case there is a bug or some edge cases, we may want to do type check and field exist check especially in python where event typing is not strongly enforced.